pkg/sidecar/attributes.go

```diff
@@ -59,7 +59,7 @@ func getResourceAttributesEnv(ns corev1.Namespace, podReferences podReferences)
 	}
 
 	envvars = append(envvars, corev1.EnvVar{
-		Name: constants.EnvPodName,
+		Name: constants.EnvPodNameSidecar,
```

Comment:
It shouldn't because I replaced its name in line 45 too.

Reply:
Is the goal behind changing this env var making it so that you could auto-instrument an opentelemetry collector?

Reply:
@iblancasa but if someone we're relying on this environment variable being set then they would no longer have it set for them. FWIW i think this change is okay, but we should maybe call it out in the changelog notes

Reply:
The problem is for the `sdk-only` injection, we are not adding any `initContainers`. As a consequence, the `isAutoInstrumentationInjected` method always returns `false`.
As a solution, I proposed to @pavolloffay to add an annotation after injecting the instrumentation. This would be really explicit and work.
But @pavolloffay suggested using the `OTEL_RESOURCE_ATTRIBUTES_POD_NAME` and/or `OTEL_RESOURCE_ATTRIBUTES_NODE_NAME` (since they are set just by the operator). But, if you also enabled the sidecar injection, those variables are already set when the pod mutator for the instrumentation injection is executed (so, it is not possible to use the presence of those variables as a guarantee that the instrumentation was injected). So @pavolloffay suggested (or at least is what I understood) to rename one of the variables for the sidecar.

That is the reason why I changed the name of the environment variable for the sidecar.

Reply:
> t, if you also enabled the sidecar injection, those variables are already set when the pod mutator for the instrumentation injection is executed (so, it is not possible to use the presence of those variables as a guarantee that the instrumentation was injected). @pavolloffay suggested (or at least is what I understood) to rename one those the variables when set in the sidecar pod mutator or in the instrumentation injection pod mutator.

The sidecar injection does not inject these variables to app container (only to the sidecar container), therefore you should be able to check for that.

Reply:
> The sidecar injection does not inject these variables to app container (only to the sidecar container), therefore you should be able to check for that.

I reviewed the source code again. Sorry, I was wrong about this.